### PR TITLE
fix(platform-node): 2.1.1 fix - update private imports and make backwards compatible

### LINKED ISSUE #606: Fix 2.1.1 _private_ import issues
- **I'm submitting a ...**
- [X] bug report
- **What modules are related to this Issue?**
- [x] platform-node

---

In the latest Core/Compiler upgrades to **2.1.1**, compiler-private & core-private have changed a few things Universal uses.

**`__compiler_private__`** looks to of been removed from `@angular/compiler`.

SelectorMatcher & CssSelector (needed for Universal), seem to be public now.

Also **`_ViewUtils`** seems to be missing / removed from `__core_private__`
It looks like it was renamed to **`view_utils`** ?

Seems to fix it:

```
import {
  SelectorMatcher as SelectorMatcher211, 
  CssSelector as CssSelector211 } from '@angular/compiler';

let SelectorMatcher;
let CssSelector;

if (SelectorMatcher211 && CssSelector211) {
  SelectorMatcher = SelectorMatcher211;
  CssSelector = CssSelector211;
} else {
  const _compiler_private_  = require('@angular/compiler').__compiler_private__;
  SelectorMatcher = _compiler_private_.SelectorMatcher;
  CssSelector = _compiler_private_.CssSelector;
}
```

ViewUtils -> view_utils can be handled in a similar way.

### PR DESCRIPTION
**Pull request by MarkPieszak**
Fixes #606

SelectorMatcher & CssSelector made public.

ViewUtils changed to view_utils

Backwards compatibility

### CODE REVIEW

File: modules/platform-node/__private_imports__.ts [hunk 1]
```diff
@@ -16,18 +18,24 @@ const {
 } = __platform_browser_private__;
 
 const {
-  ViewUtils,
+  view_utils,
```
Comment (gdi2290) [inline modules/platform-node/__private_imports__.ts]: this also needs to be backwards compatible

**Comment by John0x** [comment]
Anything new regarding this issue?

**Comment by MarkPieszak** [comment]
@gdi2290 want me to refactor this more in your way?

**Comment by qibucloud** [comment]
Angular2 在 asp.net core 里面只能这样使用？ 不能官方原生态使用？

Angular2 in asp.net core which can only be used? Can not the official use of the original ecology?

**Comment by SamD** [comment]
Any update, not seeing the fix applied. The latest npm version is 2.1.0-rc.1

**Comment by MarkPieszak** [comment]
@SamD @julienR2 I believe we need to wait for Core to make these changes, but in the meantime if you want to use 2.1.1+ you can use the `__2.1.1.workaround.ts` file: https://github.com/MarkPieszak/aspnetcore-angular2-universal/tree/master/Client here, and then include it in both the client & server bootstrap files. That can at least hold you off for now!

**Comment by julienR2** [comment]
Thanks a lot for the workaround ! I missed it during my investigations.. Not in a hurry for 2.1.1+ so far, but I keep it close in case I need a feature from this version until Core changes :)

**Comment by larsbeck** [comment]
@MarkPieszak I tried this workaround with 2.2.0, but it doesn't seem to do the trick. Should it work with 2.2.0 as well?

**Comment by MarkPieszak** [comment]
@larsbeck use the workaround fix here (https://github.com/MarkPieszak/aspnetcore-angular2-universal/tree/master/Client) take the `workaround` file put it in your project and import it to the top of your client & server main files.

`import './__2.1.1.workaround.ts'; // temporary until 2.1.1 things are patched in Core`

**Comment by larsbeck** [comment]
@MarkPieszak Ah yes, moving them to the very top fixed it. Thanks!

**Comment by gdi2290** [comment]
this gets really tricky to support semver. It's best just to follow the angular releases for at least the major and minor releases while the patches we can release out of sync

**Comment by Reuvenc** [comment]
Is there anything new regarding this PR? 
I'm trying to integrate AOT for universal, but although using the workarounds provided in universal-starter repo, I'm still getting SelectorMetcher errors (SelectorMatcher of undefined).

**Comment by gdi2290** [comment]
will be fixed in ng4 
https://github.com/angular/angular/issues/13822

**Comment by sohaibjaved44** [comment]
People who are planning to import "import './__2.1.1.workaround.ts'; " need to take care of the order as well.
Some people have mentioned that it is supposed to be placed after pollyfills but  in my case it was happy at 3rd place

**Comment by MarkPieszak** [comment]
Yes @sohaibjaved44 typically you want it below the universal polyfills, otherwise some bizarre errors pop up.

**Comment by daveriedstra** [comment]
@MarkPieszak and @sohaibjaved44 thank you! This fixed something that's been confusing me for hours. I think I was confused because of [this comment](https://github.com/aspnet/JavaScriptServices/issues/406#issuecomment-275923726).

**Pull request closed without merging**